### ISSUE TITLE
Communication channels for this community

### ISSUE THREAD

**Issue by danielballan** [bluesky/bluesky-enhancement-proposals#20]
Most communication related to this project occurs on GitHub or in the Nikea Slack. Slack has downsides for open source --- see [Why I Avoid Slack](https://matthewrocklin.com/blog/2019/02/28/slack-github) for a succinct and thorough overview. From my point of view, the most important of these are.

* On the free tier, which we use, history is not retained. (The pro tier is prohibitively expensive for this use case.)
* The conversations are not publicly searchable, and people have to "buy in" (join the Slack) before they can even passively participate. It is hard for people to discover us.

We have raised the possibility of using other platforms in addition or instead. Possibilities include a simple mailing list, a forum, or a Slack alternative. A comment by Roland Müller yesterday prompted me to re-raise it again. I'll invite him to share his perspective on the difficulties with our current platform.

**Comment by danielballan** [comment]
@tacaswell on the Friday call:

> I'd be in favor of adding a Discourse and directing any questions there that are longer than, "What is the name of this function that does X?", anything that involves sharing code snippets or long responses.

**Comment by danielballan** [comment]
@ronpandolfi voiced support for [Discord](https://discord.com/) not to be confused with _Discourse_ which is more like Slack with an emphasis on voice communication, but the free plan includes history and the paid plan is a lot cheaper than Slack's.

**Comment by ronpandolfi** [comment]
An example open-source community discord: https://discord.gg/voron

**Comment by cryos** [comment]
Does Discord help with public/searchable? It seems not, but maybe has better history. I have only used it for gaming. Discourse is a good, modern alternative to a mailing list but it does not help with the more instant/real-time discussions using text, voice, or video. The replies can be edited offering a more polished StackOverflow style capability. I would say that we might want both, i.e. somewhat ephemeral but real-time discussions and slower, well-indexed discussions that might get indexed/found.

**Comment by cryos** [comment]
GitHub also has a Discourse like feature that was announced earlier this year, and that may supplant the need for Discourse itself by enabling what I think they called Communities. It would then keep things together quite naturally with issues, development, etc while probably still wanting to consider instant communication.

**Comment by danielballan** [comment]
Yes, I think that is roughly where the conversation converged to today. We want something in the {mailng list, Discourse, GitHub Communities} space and something in the {Slack, Discord, Zulip} space.

**Comment by ronpandolfi** [comment]
@cryos Discord has its own search functionality, but it doesn't get indexed for external search. It does keep all history indefinitely, and it is internally searchable with the free plan.

This article nicely [compares Slack and Discord](https://www.elegantthemes.com/blog/resources/is-discord-the-free-slack-alternative-for-your-growing-team)

**Comment by r-m-mueller** [comment]
@danielballan the discussion on Slack alternatives (Zulip, mattermost, Rocket.Chat, gitter, discord, HipChat...) raged high at the recent European Open Science and FAIR Data collaboration and coordination [symposium](https://indico.eli-beams.eu/event/376/timetable/#all)  with an outcome of a slight preference for zulip. Personally I played with some of the alternatives. At HZB we are forced to use Mattermost, since it is provided single-sign-on by HZDR (Dresden) as a service to all Helmholtz centers. At Mattermost I miss the threading, so I have sympathy for Zulip. The mail integration could feed a mailing list for archiving purposes and passive listeners.

**Comment by danielballan** [comment]
Does anyone feel strongly that something in the {Slack, Discord, Zulip, mattermost, Rocket.Chat, HipChat, ...} space is sufficiently differentiated from Slack to be worth the switching cost? I think it's plausible that we could move, but this effort needs a "champion" to push it.

The second question, something in the {mailing list, Discourse, GitHub "Communities"}, I am inclined to champion myself. I'd like to wait and see what this new GitHub feature looks like when it emerges from beta, and then make a decision.

**Comment by danielballan** [comment]
Update: I was unable to find anything called "GitHub Communities" (mentioned offhand by @cryos above) but I think it might now be called "GitHub Discussions". I have added https://github.com/bluesky/bluesky-widgets/discussions to the Beta.

**Comment by cryos** [comment]
That is the one I meant, I remembered the name wrong or they changed it. Glad it is more widely available now.

**Comment by ronpandolfi** [comment]
Here's a neat self-hostable FOSS slack alternative: https://itsfoss.com/rocket-chat/

**Comment by danielballan** [comment]
Thanks for the link. When @r-m-mueller mentioned that in passing above it was the first I had heard of it. Does anyone have hands on experience with it?

**Comment by clintonroy** [comment]
We've been using at the Aus Synch for a little while, from a user point of view it's similar ish to slack, both a web front end and a mobile phone app that are say, 90% as polished as slacks. I'm not running it, can ask an admin how involved that is if we get to that point.

**Comment by danielballan** [comment]
I wonder if anyone is offering it as a hosted service. To borrow a good opinion from Chris Holdgraf, the ideal is an open service that you can pay someone to run for you but with the option to walk with your data and self-host at any time if the deal changes or the needs change. Not unlike email. (IIUC Discourse works like this too.)

**Comment by r-m-mueller** [comment]
> I wonder if anyone is offering it as a hosted service. To borrow a good opinion from Chris Holdgraf, the ideal is an open service that you can pay someone to run for you but with the option to walk with your data and self-host at any time if the deal changes or the needs change. Not unlike email. (IIUC Discourse works like this too.)

This is probably worth to be considered within the 5 Light Source (5LS) Remote Experiment Forum. Personal experiences I could contribute are: 
(1) Within Helmholtz federated IT services HZDR(Dresden) provides mattermost, FZJ(Jülich) provides Rocket.Chat to all collaborating Helmholtz centers, s. [roadmap](https://www.hifis.net/roadmap/). At HZB I exported Slack workspaces to mattermost hosted on HZDR premises. 
(2) Heinz Junkes FHI/MPG hosts Rocket.Chat on FHI premises.

Authentication, access to the hosted services generate specific hurdles: (1) uses the Helmholtz wide AAI, (2) the FHI/MPG AAI. As HZB member I have immediate access to (1) e.g. mattermost, like 1257 other Helmholtz employees today. Since public channels are open for so many others, not familiar with the topic, one tends to use private channels. That is hampering the intended communication. For (2) I am alien, so I needed an invitation from the FHI inside, in all cases of trouble I need to contact an unknown admin of the hosting organisation.

Personal summary: comfort of an "open" hosted service comes with significant constraints. These are beyond the limitations of the tool itself.
